tests/test_oi.py

```diff
@@ -0,0 +1,124 @@
+from devito.jit_manager import JitManager
+import numpy as np
+from devito.propagator import Propagator
+from sympy import IndexedBase, symbols, Eq
+from sympy.abc import x, y, z, t
+
+
+class Test_OI_Calculation(object):
+    g = None
+
+    def test_oi_1(self):
+        '''OI = (ADD+MUL)/[(LOAD+STORE)*word_size]; word_size=8(double),4(float)
+        Equation = v1[i2][i1] = 3*v2[i2][i1] + 2*v3[i2][i1];
+        '''
+        load = 3.0
+        store = 1.0
+        add = 1.0
+        mul = 2.0
+        dtype = np.float32
+        i1, i2 = symbols('i1 i2')
+        eq = Eq(x(i2, i1), 3*y(i2, i1) + 2*t(i2, i1))
+        data = np.arange(50, dtype=np.float32).reshape((10, 5))
+        v1 = IndexedBase('v1')
+        v2 = IndexedBase('v2')
+        v3 = IndexedBase('v3')
+
+        propagator = Propagator("process", 10, (5,), 0)
+        propagator.stencils = (eq,)
+        propagator.stencil_args = ([v1[i2, i1], v2[i2, i1], v3[i2, i1]], )
+        propagator.subs = (x(i2, i1), y(i2, i1), t(i2, i1))
+        propagator.add_param("v1", data.shape, data.dtype)
+        propagator.add_param("v2", data.shape, data.dtype)
+        propagator.add_param("v3", data.shape, data.dtype)
+
+        if self.g is None:
+            self.g = JitManager([propagator])
+        else:
+            self.g.function_descriptors = [propagator]
+        f = self.g.get_wrapped_functions()[0]
+
+        propagator_oi = propagator.get_kernel_oi(dtype)
+        hand_oi = (mul+add)/((load+store)*np.dtype(dtype).itemsize)
```

Comment:
As mentioned in the general comment, `hand_oi` assumes no caching, equivalent to `oi_low`. We either need to test  both `oi_high` (optimal cache) and `oi_low` (pessimal cache) or drop `oi_low` altogether. The same goes for the other tests.